src/adapter/src/coord/sequencer/cluster.rs

```diff
@@ -671,30 +676,37 @@ impl Coordinator {
                     name,
                     &compute,
                     new_size,
-                    &mut ops,
+                    &mut create_ops,
                     // AVAILABILITY ZONES hasn't changed, so existing replicas don't need to be
                     // rescheduled.
                     Some(new_availability_zones.as_ref()),
                     *new_disk,
+                    new_replica_start_pending,
                     owner_id,
                     reason.clone(),
                 )?;
                 create_cluster_replicas.push((cluster_id, id))
             }
         }
 
-        let variant = ClusterVariant::Managed(new_config);
-        ops.push(catalog::Op::UpdateClusterConfig {
-            id: cluster_id,
-            name,
-            config: ClusterConfig { variant },
-        });
-
-        self.catalog_transact(session, ops).await?;
+        // we only finalize if we created something and the thing we created
+        // started as pending
+        let mut needs_finalization = true;
+        if create_cluster_replicas.is_empty() || !new_replica_start_pending {
```

Comment:
I'm dubious about the logic condition here and would like something simpler if possible. For example, this came to mind: is there some scenario where this if block evals to `false`, but `drop_ops` contains elements? In that case the `drop_ops` would not be executed, which seems almost certainly wrong. Should this instead be maybe `if !drop_ops.is_empty()`? Is this if block needed at all?

Reply:
I've rewritten this to attempt to make it more clear, but I think it does work as written.

If we require finalization we do not drop or update the cluster config, regardless of what drop_ops contains. The only odd feature of this is that if we resize and lower the replication factor no original replicas will be dropped until finalization. I'm ok with that.